### DISCUSSION TITLE
[@types/jsonwebtoken] Request to narrow types reference for node

### DISCUSSION THREAD

**Discussion by shawnmcknight** [DefinitelyTyped/DefinitelyTyped#64543] [Issues with a @types package]
Currently the types for this library have the following triple slash directive, which brings in the entire set of typings for `@types/node`:
```ts
/// <reference types="node" />
```

However, referencing all of `@types/node` can have unanticipated side effects as it might bring the ambient declaration for all of the node types into scope.  See this [gist](https://gist.github.com/RyanCavanaugh/702ebd1ca2fc060e58e634b4e30c1c1c) for a pretty good explainer of this issue from the TypeScript team.

I understand that this lib is intended for node environments, but it would be good to narrow the scope of how much of `@types/node` is being brought in to prevent unintended side effects.  My request is to replace the triple slash directive above with:
```
/// <reference types="node/buffer" />
/// <reference types="node/crypto" />
```

which will target only the subsets of `@types/node` that are currently being used by this declaration file.  It should continue to work as expected but not bring in the entire ambient declaration for `@types/node`.

Thanks!

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
@shawnmcknight I'd love to bring this to fruition, but I cannot make this working using existing DT setup. The CI tests would eventually fail, as the type resolution for those scoped imports forces lookup for @types/node within 'node_modules' for some reason.
I'd suggest to bring a question, if that is still a valid approach (a lot changed since 2020, TS at DT is now at 5.1 version) on the TS Discord channel here:
https://discord.com/channels/508357248330760243/757992230077333655
reference me, pls in the topic
